Assets/Scripts/Managers/PlayerDeathHandler.cs

```diff
@@ -68,6 +68,7 @@ void Update()
                     {
                         destroyed.GetComponent<SpriteRenderer>().enabled = true;
                         destroyed.GetComponent<Collider2D>().enabled = true;
+                        destroyed.layer = LayerMask.NameToLayer("Default");
```

Comment:
The destroyer bullet changes the object's layer to "Destroyed," so that line is needed to reset this change. Not doing this is what caused the player to go through the blocks.